File: 2020/julia/src/Day1.jl
module Day1

include("Aoc.jl")
using .Aoc

amounts = parse.(Int64, Aoc.input_lines(1))
        
function part1()
    for a in amounts, b in amounts
        if a + b == 2020
            return a * b
        end
    end
end

function part2()
    for a in amounts, b in amounts, c in amounts
        if a + b + c == 2020
            return a * b * c
        end
    end
end

end

File: 2020/typescript/src/days/day1.ts
import {Day} from "../aoc";

export class Day1 extends Day {
    private amounts: number[] = [];

    day = (): number => 1;

    part1(): string {
        for (const a of this.amounts) {
            for (const b of this.amounts) {
                if (a + b === 2020) return (a * b).toString();
            }
        }

        return '';
    }

    part2(): string {
        for (const a of this.amounts) {
            for (const b of this.amounts) {
                for (const c of this.amounts) {
                    if (a + b + c === 2020) return (a * b * c).toString();
                }
            }
        }

        return '';
    }

    setup() {
        super.setup();

        this.amounts = this.input.map((i) => parseInt(i));
    }
}